api/errors.go

```diff
@@ -10,39 +11,96 @@ import (
 // The canonical errors from the EigenDA gRPC API endpoints.
 //
 // Notes:
-// - We start with a small (but sufficient) subset of grpc's error codes,
-//   and expand when there is an important failure case to separate out. See:
-//   https://grpc.io/docs/guides/status-codes/
-// - Make sure that internally propagated errors are eventually wrapped in one of the
-//   user-facing errors defined here, since grpc otherwise returns an UNKNOWN error code,
-//   which is harder to debug and understand for users.
-// - See https://github.com/googleapis/googleapis/blob/ba8ea80f25d19bde8501cd51f314391f8d39bde8/google/rpc/code.proto
-//   for the mapping of grpc error codes to HTTP status codes.
-
+//   - We start with a small (but sufficient) subset of grpc's error codes,
+//     and expand when there is an important failure case to separate out. See:
+//     https://grpc.io/docs/guides/status-codes/
+//   - Make sure that internally propagated errors are eventually wrapped in one of the
+//     user-facing errors defined here, since grpc otherwise returns an UNKNOWN error code,
+//     which is harder to debug and understand for users.
+//   - See https://github.com/googleapis/googleapis/blob/ba8ea80f25d19bde8501cd51f314391f8d39bde8/google/rpc/code.proto
+//     for the mapping of grpc error codes to HTTP status codes.
 func newErrorGRPC(code codes.Code, msg string) error {
 	return status.Error(code, msg)
 }
 
+// WrapAsGRPCError wraps an error with a gRPC code and message. If the error already has a gRPC code,
+// it retains the code but combines the messages. Otherwise, it uses the provided code and combined message.
+func WrapAsGRPCError(err error, code codes.Code, msg string) error {
+	if err == nil {
+		return nil
+	}
+
+	combinedMessage := fmt.Sprintf("%s: %s", msg, err.Error())
+	if s, ok := status.FromError(err); ok {
+		// If it's already a gRPC error, use the same code but combine the messages
+		return status.Error(s.Code(), combinedMessage)
```

Comment:
why use the old code? Why not the new one?

Reply:
explained above

Reply:
This doesn't match your description above though?
If externalService returns 400, we want to return 500 to our users, not 400. So you should use the code passed to wrapAsGRPCError always I think?

Reply:
The above explanation is basically "if lower level function has opinion of error, preserve it"

Reply:
But that is just wrong...